tutorials/plot_receptive_field.py

```diff
@@ -0,0 +1,227 @@
+# -------------------------------------------
+#
+# Finally, we'll use the :class:`mne.decoding.ReceptiveField` class to recover
+# the linear receptive field of this signal. Note that properties of the
+# receptive field (e.g. smoothness) will depend on the autocorrelation in the
+# inputs and outputs.
+
+# Create training and testing data
+train, test = np.arange(20), 21
+X_train, X_test, y_train, y_test = X[train], X[test], y[train], y[test]
+X_train, X_test, y_train, y_test = [np.rollaxis(ii, -1, 0) for ii in
+                                    (X_train, X_test, y_train, y_test)]
+# Model the simulated data as a function of the spectrogram input
+alphas = np.logspace(-4, 0, 10)
+scores = np.zeros_like(alphas)
+models = []
+for ii, alpha in enumerate(alphas):
+    mod = ReceptiveField(tmin, tmax, sfreq, freqs, model=Ridge(alpha))
+    mod.fit(X_train, y_train)
+
+    # Now make predictions about the model output, given input stimuli.
+    y_pred = mod.predict(X_test).squeeze()
+    scores[ii] = r2_score(y_pred, y_test[mod.mask_predict_])
+    models.append(mod)
+delay_sec_mod = mod.delays_ / sfreq
+
+# Choose the model that performed best on the held out data
+ix_best_alpha = np.argmax(scores)
+best_mod = models[ix_best_alpha]
+coefs = best_mod.coef_
+best_pred = best_mod.predict(X_test).squeeze()
+
+# Plot the original STRF, and the one that we recovered with modeling.
+fig, (ax1, ax2) = plt.subplots(1, 2, figsize=(10, 5), sharey=True, sharex=True)
+ax1.pcolormesh(delays_sec, freqs, weights, cmap='RdBu_r')
+ax2.pcolormesh(delay_sec_mod, mod.feature_names, coefs, cmap='RdBu_r')
+ax1.set_title('Original STRF')
+ax2.set_title('Best Reconstructed STRF')
+plt.setp([iax.get_xticklabels() for iax in [ax1, ax2]], rotation=45)
+plt.autoscale(tight=True)
+mne.viz.tight_layout()
+
+# Plot the actual response and the predicted response on a held out stimulus
+time_pred = np.arange(y_pred.shape[0]) / sfreq
+fig, ax = plt.subplots()
+ax.plot(time_pred, y_test[mod.mask_predict_], color='k', alpha=.2, lw=4)
+ax.plot(time_pred, best_pred, color='r', lw=1)
+ax.set(title='Original and predicted activity', xlabel='Time (s)')
+ax.legend(['Original', 'Predicted'])
+plt.autoscale(tight=True)
+mne.viz.tight_layout()
+
+
+###############################################################################
+# Visualize the effects of regularization
+# ---------------------------------------
+#
+# Above we fit a :class:`mne.decoding.ReceptiveField` model for one of many
+# values for the "ridge" parameter. Here we will plot the model score as well
+# as the model coefficients for each value, in order to visualize how
+# coefficients change with different levels of regularization. These issues
+# as well as the STRF pipeline are described in detail in [1]_ and [2]_
+
+# Plot model score for each ridge parameter
+fig = plt.figure(figsize=(20, 4))
+ax = plt.subplot2grid([2, 10], [1, 0], 1, 10)
+ax.scatter(np.arange(len(alphas)), scores, s=40, color='r')
```

Comment:
Since this is considered a continuum of values that you are sampling well enough to approximate the curve, `ax.plot(..., marker='o'` would probably be better